sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -111,6 +147,55 @@ private static PipelineResult run(
     return result;
   }
 
+  private static PipelineResult limitRun(
```

Comment:
The point of `run` was to have exactly one for everything. Seeing as that isn't going to work, you should probably just inline this.

Reply:
I want to use this function to encapsulate related logic together.

Reply:
To avoid a long `limitCollect` function.